File: pages/dailypaper.js
(function () {
  const response = {
    "type": "page",
    "body": [
      {
        "type": "crud",
        "syncLocation": false,
        "resizable": true,
        "columnsTogglable": true,
        "className": "table-db m-b-none",
        "tableClassName": "table-cell-nowrap",
        "api": {
          "url": "/daily_paper/daily_paper_list",
          "method": "get",
          "dataType": "json",
          "adaptor": function (payload) {
            return {
              status: payload.status,
              msg: payload.msg,
              data: {
                items: payload.data && payload.data.items ? payload.data.items : [],
                total: payload.data && payload.data.total ? payload.data.total : 0
              }
            };
          }
        },
        "headerToolbar": [
          {
            "type": "tpl",
            "tpl": "<div class='header-title'><i class='fa fa-file-text'></i> 盾构日报表数据</div>"
          },
          "export-excel",
          {
            "type": "reload",
            "icon": "fa fa-refresh",
            "tooltip": "刷新数据",
            "tooltipPlacement": "top",
            "level": "link"
          }
        ],
        "exportExcel": {
          "filename": "日报表数据",
          "columns": [
            "id", "areaId", "ljsh", "dpnj", "jjbz", "cjll", "dpzs", "kslc", "qjcd",
            "tjsd", "jjll", "cjbz", "jrjd", "ztl", "qjyl", "grd", "wcd",
            "qkyl", "qpcyl", "qjmc", "dgjbh", "pjrjc", "jslc", "sfsj", "time"
          ]
        },
        "filter": {
          "title": "查询条件",
          "submitText": "查询",
          "body": [
            {
              "type": "input-text",
              "name": "id",
              "label": "编号",
              "placeholder": "请输入编号进行查询"
            },
            {
              "type": "input-text",
              "name": "areaId",
              "label": "盾构机编码id",
              "placeholder": "请输入盾构机编码id进行查询"
            },
            {
              "type": "input-text",
              "name": "qjmc",
              "label": "区间名称",
              "placeholder": "请输入区间名称进行查询"
            },
            {
              "type": "input-text",
              "name": "dgjbh",
              "label": "盾构机编号",
              "placeholder": "请输入盾构机编号进行查询"
            }
          ]
        },
        "columns": [
          {
            "type": "operation",
            "label": "操作",
            "width": 80,
            "fixed": "right",
            "buttons": [
              {
                "type": "button",
                "icon": "fa fa-pencil",
                "tooltip": "修改",
                "level": "link",
                "actionType": "drawer",
                "drawer": {
                  "position": "right",
                  "size": "lg",
                  "title": "修改日报表信息",
                  "body": {
                    "type": "form",
                    "api": {
                      "method": "post",
                      "url": "/daily_paper/update_daily_paper",
                      "messages": {
                        "success": "更新成功",
                        "failed": "更新失败"
                      }
                    },
                    "body": [
                      {"type": "hidden", "name": "id"},
                      {"type": "input-number", "name": "areaId", "label": "盾构机编码id"},
                      {"type": "input-number", "name": "ljsh", "label": "累积时间"},
                      {"type": "input-text", "name": "dpnj", "label": "刀盘扭矩"},
                      {"type": "input-text", "name": "jjbz", "label": "进尺比重"},
                      {"type": "input-text", "name": "cjll", "label": "出浆流量"},
                      {"type": "input-text", "name": "dpzs", "label": "刀盘转速"},
                      {"type": "input-number", "name": "kslc", "label": "开始里程"},
                      {"type": "input-number", "name": "qjcd", "label": "区间长度"},
                      {"type": "input-text", "name": "tjsd", "label": "推进速度"},
                      {"type": "input-text", "name": "jjll", "label": "进浆流量"},
                      {"type": "input-text", "name": "cjbz", "label": "出浆比重"},
                      {"type": "input-text", "name": "jrjd", "label": "今日进度环"},
                      {"type": "input-text", "name": "ztl", "label": "总推力"},
                      {"type": "input-number", "name": "qjyl", "label": "区间余量"},
                      {"type": "input-text", "name": "grd", "label": "贯入度"},
                      {"type": "input-text", "name": "wcd", "label": "完成度"},
                      {"type": "input-text", "name": "qkyl", "label": "初压力"},
                      {"type": "input-text", "name": "qpcyl", "label": "气泡仓压力"},
                      {"type": "input-text", "name": "qjmc", "label": "区间名称"},
                      {"type": "input-text", "name": "dgjbh", "label": "盾构机编号"},
                      {"type": "input-number", "name": "pjrjc", "label": "平均日进尺"},
                      {"type": "input-number", "name": "jslc", "label": "结束里程"},
                      {"type": "input-text", "name": "sfsj", "label": "始发时间"},
                      {"type": "input-datetime", "name": "time", "label": "日期"}
                    ]
                  }
                }
              },
              {
                "type": "button",
                "icon": "fa fa-times text-danger",
                "actionType": "ajax",
                "tooltip": "删除",
                "confirmText": "确定要删除该记录吗？",
                "api": {
                  "method": "delete",
                  "url": "/daily_paper/delete_daily_paper/${id}"
                },
                "messages": {
                  "success": "删除成功",
                  "failed": "删除失败"
                }
              }
            ]
          },
          { "name": "id", "label": "编号", "type": "number", "width": 80 },
          { "name": "areaId", "label": "盾构机编码id", "type": "number", "width": 120 },
          { "name": "ljsh", "label": "累积时间", "type": "number", "width": 100 },
          { "name": "dpnj", "label": "刀盘扭矩", "type": "text", "width": 100 },
          { "name": "jjbz", "label": "进尺比重", "type": "text", "width": 100 },
          { "name": "cjll", "label": "出浆流量", "type": "text", "width": 100 },
          { "name": "dpzs", "label": "刀盘转速", "type": "text", "width": 100 },
          { "name": "kslc", "label": "开始里程", "type": "number", "width": 100 },
          { "name": "qjcd", "label": "区间长度", "type": "number", "width": 100 },
          { "name": "tjsd", "label": "推进速度", "type": "text", "width": 100 },
          { "name": "jjll", "label": "进浆流量", "type": "text", "width": 100 },
          { "name": "cjbz", "label": "出浆比重", "type": "text", "width": 100 },
          { "name": "jrjd", "label": "今日进度环", "type": "text", "width": 100 },
          { "name": "ztl", "label": "总推力", "type": "text", "width": 100 },
          { "name": "qjyl", "label": "区间余量", "type": "number", "width": 100 },
          { "name": "grd", "label": "贯入度", "type": "text", "width": 100 },
          { "name": "wcd", "label": "完成度", "type": "text", "width": 100 },
          { "name": "qkyl", "label": "初压力", "type": "text", "width": 100 },
          { "name": "qpcyl", "label": "气泡仓压力", "type": "text", "width": 100 },
          { "name": "qjmc", "label": "区间名称", "type": "text", "width": 100 },
          { "name": "dgjbh", "label": "盾构机编号", "type": "text", "width": 100 },
          { "name": "pjrjc", "label": "平均日进尺", "type": "number", "width": 100 },
          { "name": "jslc", "label": "结束里程", "type": "number", "width": 100 },
          { "name": "sfsj", "label": "始发时间", "type": "text", "width": 100 },
          { "name": "time", "label": "日期", "type": "date", "width": 120 }
        ],
        "perPage": 10,
        "messages": {
          "fetchFailed": "查询失败，请检查网络或联系管理员"
        },
        "style": {
          "table": {
            "overflow": "auto",
            "maxHeight": "500px"
          }
        },
        "cssVars": {
          "--Table-content-paddingX": "1rem",
          "--Table-thead-height": "2.5rem"
        },
        "css": {
          ".text-ellipsis": {
            "whiteSpace": "nowrap",
            "overflow": "hidden",
            "textOverflow": "ellipsis",
            "maxWidth": "100%"
          },
          ".table-db th": {
            "backgroundColor": "#f5f7fa",
            "fontWeight": "500"
          },
          ".table-db td": {
            "padding": "8px"
          }
        }
      }
    ]
  };

  // 添加自定义样式
  const style = document.createElement('style');
  style.textContent = `
    .text-ellipsis {
      white-space: nowrap !important;
      overflow: hidden !important;
      text-overflow: ellipsis !important;
      max-width: 100% !important;
    }
    .text-nowrap {
      white-space: nowrap !important;
    }
    .table-db {
      border-radius: 4px;
      overflow: hidden;
    }
    .table-db th {
      background-color: #f5f7fa !important;
      font-weight: 500;
    }
    .table-db td {
      max-width: 0 !important;
      padding: 8px !important;
      overflow: hidden !important;
    }
    .table-db .cxd-Table-content {
      overflow: auto !important;
    }
  `;
  document.head.appendChild(style);

  window.jsonpCallback && window.jsonpCallback(response);
})();

File: api/ORM/api/daily_paper.py
from fastapi import APIRouter, Query, HTTPException, Depends
from typing import Optional
from daily_models import daily_paper
from tortoise.expressions import Q
from pydantic import BaseModel
from Rbac_utils import get_user_from_token, check_permission

daily_paper_api = APIRouter()

class DailyPaperUpdateModel(BaseModel):
    id: int
    areaId: Optional[int] = None
    ljsh: Optional[float] = None
    dpnj: Optional[str] = None
    jjbz: Optional[str] = None
    cjll: Optional[str] = None
    dpzs: Optional[str] = None
    kslc: Optional[float] = None
    qjcd: Optional[float] = None
    tjsd: Optional[str] = None
    jjll: Optional[str] = None
    cjbz: Optional[str] = None
    jrjd: Optional[str] = None
    ztl: Optional[str] = None
    qjyl: Optional[float] = None
    grd: Optional[str] = None
    wcd: Optional[str] = None
    qkyl: Optional[str] = None
    qpcyl: Optional[str] = None
    qjmc: Optional[str] = None
    dgjbh: Optional[str] = None
    pjrjc: Optional[float] = None
    jslc: Optional[float] = None
    sfsj: Optional[str] = None
    time: Optional[str] = None

@daily_paper_api.get("/daily_paper_list")
async def get_daily_paper_list(
    id: Optional[int] = Query(None, description="编号"),
    areaId: Optional[int] = Query(None, description="盾构机编码id"),
    qjmc: Optional[str] = Query(None, description="区间名称"),
    dgjbh: Optional[str] = Query(None, description="盾构机编号"),
    page: int = Query(1, ge=1, description="页码"),
    per_page: int = Query(10, ge=1, le=100, description="每页数量")
):
    """获取盾构日报表数据（支持多条件筛选、分页）"""
    
    # 构建查询条件
    filters = Q()
    if id is not None:
        filters &= Q(id=id)
    if areaId is not None:
        filters &= Q(areaId=areaId)
    if qjmc is not None:
        filters &= Q(qjmc=qjmc)
    if dgjbh is not None:
        filters &= Q(dgjbh=dgjbh)

    # 获取总记录数
    total = await daily_paper.filter(filters).count()

    # 分页查询数据
    items = await daily_paper.filter(filters)\
        .offset((page - 1) * per_page)\
        .limit(per_page)\
        .order_by("-time")\
 
    return {
        "status": 0,
        "msg": "ok",
        "data": {
            "items": [{
                "id": item.id,
                "areaId": item.areaId,
                "ljsh": item.ljsh,
                "dpnj": item.dpnj,
                "jjbz": item.jjbz,
                "cjll": item.cjll,
                "dpzs": item.dpzs,
                "kslc": item.kslc,
                "qjcd": item.qjcd,
                "tjsd": item.tjsd,
                "jjll": item.jjll,
                "cjbz": item.cjbz,
                "jrjd": item.jrjd,
                "ztl": item.ztl,
                "qjyl": item.qjyl,
                "grd": item.grd,
                "wcd": item.wcd,
                "qkyl": item.qkyl,
                "qpcyl": item.qpcyl,
                "qjmc": item.qjmc,
                "dgjbh": item.dgjbh,
                "pjrjc": item.pjrjc,
                "jslc": item.jslc,
                "sfsj": item.sfsj,
                "time": item.time,
            } for item in items],
            "total": total,
            "page": page,
            "per_page": per_page
        }
    }

@daily_paper_api.post("/update_daily_paper")
async def update_daily_paper(paper_data: DailyPaperUpdateModel):
    """更新日报表数据"""
    try:
        paper = await daily_paper.get_or_none(id=paper_data.id)
        if not paper:
            return {"status": 1, "msg": f"未找到ID为{paper_data.id}的记录"}
        
        await paper.update_from_dict(paper_data.dict(exclude_unset=True))
        await paper.save()
        return {"status": 0, "msg": "更新成功"}
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"更新失败: {str(e)}")

@daily_paper_api.delete("/delete_daily_paper/{id}")
async def delete_daily_paper(id: int):
    """删除日报表数据"""
    try:
        paper = await daily_paper.get_or_none(id=id)
        if not paper:
            return {"status": 1, "msg": f"未找到ID为{id}的记录"}
        
        await paper.delete()
        return {"status": 0, "msg": "删除成功"}
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"删除失败: {str(e)}")

